### DISCUSSION TITLE
Custom templates

### DISCUSSION THREAD

**Discussion by andsouth44** [aristanetworks/avd#3142] [Q&A] [answered]
I would like to contribute some documentation about Custom Templates. Maybe a "How to Use Custom Templates" guide document or enhance the existing documentation pages somehow.
Just wanted to check that nobody is working on this already?
And what format would be preferred?

**Comment by ClausHolbechArista** [comment]
Hi Andrew

Thank you very much for offering this. No one else is working on this, so your contribution is most welcome.

We would prefer the guide as a markdown file and submitted in a pull request. Depending on which type of custom template we are talking about here (`eos_designs` or `eos_cli_config_gen`), the file should be placed in a `docs/how-to` folder under the role. To get the guide into the table of contents it should also be added to the `mkdocs.yml` file. We have how-tos for `eos_designs` and recently for `eos_config_deploy_cvp`, so please follow the style of those.

We are here to help if you need assistance.

Thanks again and looking forward to reviewing.

BR
Claus

**Reply by andsouth44** [reply, marked as answer]
Thanks Claus.
I will give that a try.
